### ISSUE TITLE
gnuradio does not build from source, not finding pygtk

### ISSUE THREAD

**Issue by natevw** [Homebrew/homebrew#44813]
I have done:

```
brew install --build-from-source pygtk
# succeeds, no caveats/warnings, pygtk-2.24.0_1

brew install --build-from-source gnuradio
# fails, because pygtk not found :-/
```

https://gist.github.com/anonymous/58cff8a359c7f06e5dc3#file-05-cmake-L314

**Comment by natevw** [comment]
It looks like the check for this in gnuradio itself is implemented here: https://github.com/gnuradio/gnuradio/blob/68c6862388827406bf7e3ad778b9627ef525921a/grc/CMakeLists.txt#L28

So I ran `brew sh` and inside python there, was able to `import gtk` and verify that `gtk.pygtk_version >= (2, 10, 0)` — but for some reason during the build it still thinks PyGTK is not available.

**Comment by natevw** [comment]
Using `brew install -d --build-from-source gnuradio`, when it fails I am able to get a shell and test the python environment there. Indeed, `import gtk` fails under those circumstances.

I started my working system python, and the broken homebrew build python, and the main difference is that the system python starts up including an `import sitecustomize` module, which the brew build python doesn't:

<img width="1530" alt="screen shot 2015-10-10 at 8 42 41 am" src="https://cloud.githubusercontent.com/assets/265902/10411944/2c5095bc-6f2b-11e5-98a7-7c0bc8382ff3.png">

Why would these startup sequences differ?

**Comment by natevw** [comment]
Looks like the `site.USER_SITE` is "/private/tmp/gnuradio20151010-48744-f79hiy/gnuradio-3.7.8/.brew_home/Library/Python/2.7/lib/python/site-packages" in the build python and in the "normal" python it is "/Users/natevw/Library/Python/2.7/lib/python/site-packages".

Since the user site file is where we are encouraged to put homebrew.pth, I suspect this is the problem. Where this is getting set I'm not sure, and I wouldn't be surprised if it is somewhat intentional — but it seems to mean that brew-installed Python dependencies do not show up.

**Comment by natevw** [comment]
As a workaround, I tried to make a new file at /System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/sitecustomize.py, containing:

```
import site; site.addsitedir("/usr/local/lib/python2.7/site-packages")
```

(This is what homebrew originally had me put in ~/Library/Python/2.7/lib/python/site-packages/homebrew.pth instead.)

This gets the homebrew packages into play at the system level, rather than user level, but *still* doesn't work! Somehow homebrew is making sure that Python does not use its own brewed packages…??!

**Comment by natevw** [comment]
Finally! I noticed that in the broken build python shell, `import sitecustomize; sitecustomize.__file__` was somehow pointing to "/usr/local/lib/python2.7/site-packages/sitecustomize.py" rather than the one next to `site.__file__`.

So I added that to add:

```
else:
    import site; site.addsitedir("/usr/local/lib/python2.7/site-packages")
```
as an alternate to the `if os.path.realpath(sys.executable).startswith('/usr/local/Cellar/python')` logic.

So with that hack and the patch in https://github.com/Homebrew/homebrew/issues/44811, I have a working gnuradio again, after naïvely running `brew upgrade` (which should have been `brew upgrade --build-from-source` at least, although I'm not sure that would have avoided this either…I had originally installed via @metacollin's tap).

(Note that I still need the file at /System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/sitecustomize.py or in home directory mentioned in previous comment for homebrew deps to work from elsewhere.)
